setTimeout and performance.now are not defined when using rnx-start

### What happened?

My app crahses with the following
<details>
<summary>App Crash</summary>
-23 11:29:40.853 15148 15378 I HermesVM:     at anonymous (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:37223:133)
06-23 11:29:40.853 15148 15378 I HermesVM:     at loadModuleImplementation (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:328:14)
06-23 11:29:40.853 15148 15378 I HermesVM:     at guardedLoadModule (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:227:38)
06-23 11:29:40.853 15148 15378 I HermesVM:     at metroRequire (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:123:92)
06-23 11:29:40.853 15148 15378 I HermesVM:     at anonymous (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:59400:124)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: Exception in native call from JS
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: com.facebook.react.devsupport.JSException: Cannot read property 'now' of undefined
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: 	at com.facebook.jni.NativeRunnable.run(Native Method)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: 	at android.os.Handler.handleCallback(Handler.java:942)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: 	at android.os.Handler.dispatchMessage(Handler.java:99)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: 	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: 	at android.os.Looper.loopOnce(Looper.java:201)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: 	at android.os.Looper.loop(Looper.java:288)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: 	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:228)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: 	at java.lang.Thread.run(Thread.java:1012)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: Caused by: com.facebook.jni.CppException: Cannot read property 'now' of undefined
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: 
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: TypeError: Cannot read property 'now' of undefined
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at anonymous (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:37223:133)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at loadModuleImplementation (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:328:14)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at guardedLoadModule (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:227:38)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at metroRequire (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:123:92)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at anonymous (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:59400:124)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at loadModuleImplementation (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:328:14)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at guardedLoadModule (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:227:38)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at metroRequire (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:123:92)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at get AppRegistry (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:1592:25)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at FirebaseMessagingModule (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:278594:19)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at firebaseModuleWithApp (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:281318:74)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at anonymous (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:278282:52)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at loadModuleImplementation (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:328:14)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at guardedLoadModule (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:227:38)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at metroRequire (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:123:92)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at anonymous (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:123987:14)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at loadModuleImplementation (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:328:14)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at guardedLoadModule (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:227:38)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at metroRequire (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:123:92)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at anonymous (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:1465:104)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at loadModuleImplementation (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:328:14)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at guardedLoadModule (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:219:47)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at metroRequire (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:123:92)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative:     at global (http://10.0.2.2:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true:790292:4)
06-23 11:29:40.856 15148 15378 E unknown:ReactNative: 	... 8 more
06-23 11:29:40.856 15148 15378 E ReactNativeJNI: Attempting to call JS function on a bad application bundle: HMRClient.setup()
</details>

It works fine with standard metro.

I've compared the bundles and the diff is
```diff
--- cli.js	2023-06-23 11:30:19.728315729 +1000
+++ rnx.js	2023-06-23 11:29:56.764407628 +1000
@@ -790289,7 +790289,6 @@
 }
 ;
 },6110,[],"app.json");
-__r(53);
 __r(0);
 //# sourceMappingURL=http://127.0.0.1:8081/index.map?platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true
 //# sourceURL=http://127.0.0.1:8081/index.bundle//&platform=android&dev=true&minify=false&app=com.gladlyapp.mobile.debug&modulesOnly=false&runModule=true
```

53 is 
```
},52,[53],"node_modules/react-native/Libraries/ReactPrivate/ReactNativePrivateInitializeCore.js");
```

which is a pretty critical file! :)

### Affected Package

@rnx-kit/cli or perhaps the metro config?

### Version

0.16.9

### Which platforms are you seeing this issue on?

- [X] Android
- [ ] iOS
- [ ] macOS
- [ ] Windows

### System Information

```true
System:
  OS: Linux 6.3 Ubuntu 22.10 22.10 (Kinetic Kudu)
  CPU: (32) x64 AMD Ryzen 9 7950X 16-Core Processor
  Memory: 36.22 GB / 61.95 GB
  Shell:
    version: 5.2.2
    path: /bin/bash
Binaries:
  Node:
    version: 18.16.0
    path: ~/.nvm/versions/node/v18.16.0/bin/node
  Yarn:
    version: 1.22.19
    path: ~/.nvm/versions/node/v18.16.0/bin/yarn
  npm:
    version: 9.6.7
    path: ~/.nvm/versions/node/v18.16.0/bin/npm
  Watchman:
    version: 4.9.0
    path: /usr/bin/watchman
SDKs:
  Android SDK:
    API Levels:
      - "33"
      - "34"
    Build Tools:
      - 28.0.3
      - 29.0.2
      - 29.0.3
      - 30.0.2
      - 30.0.3
      - 31.0.0
      - 31.0.0
      - 32.1.0
      - 33.0.0
      - 33.0.0
      - 34.0.0
      - 34.0.0
    System Images:
      - android-33 | Google APIs Intel x86 Atom_64
      - android-33 | Google Play Intel x86 Atom_64
    Android NDK: Not Found
IDEs:
  Android Studio: AI-213.7172.25.2113.9123335
Languages:
  Java:
    version: 11.0.19
    path: /usr/bin/javac
  Ruby:
    version: 3.0.4
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.0
    wanted: ^0.72.0
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: Not found
```


### Steps to Reproduce

* Init a new react native 0.72.0 app
* Add rnx-kit
* npm run react-native rnx-start
* Run the app

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/.changeset/red-balloons-arrive.md
+++ b/.changeset/red-balloons-arrive.md
@@ -1,0 +1,5 @@
+---
+"@rnx-kit/metro-resolver-symlinks": patch
+---
+
+Correctly handle redirected modules

--- a/packages/metro-resolver-symlinks/src/symlinkResolver.ts
+++ b/packages/metro-resolver-symlinks/src/symlinkResolver.ts
@@ -16,13 +16,20 @@ import { remapImportPath } from "./utils/remapImportPath";
 
 function applyMetroResolver(
   resolve: CustomResolver,
-  context: ResolutionContext,
+  ctx: ResolutionContext,
   moduleName: string,
   platform: string
 ): Resolution {
-  const modifiedModuleName = resolveModulePath(context, moduleName, platform);
+  // Resolve redirects before we try to resolve the module:
+  // https://github.com/facebook/metro/blob/v0.76.7/docs/Resolution.md#redirectmodulepath-string--string--false
+  const realModuleName = ctx.redirectModulePath(moduleName);
+  if (realModuleName === false) {
+    return { type: "empty" };
+  }
+
+  const modifiedModuleName = resolveModulePath(ctx, realModuleName, platform);
   // @ts-expect-error We pass 4 arguments instead of 3 to be backwards compatible
-  return resolve(context, normalizePath(modifiedModuleName), platform, null);
+  return resolve(ctx, normalizePath(modifiedModuleName), platform, null);
 }
 
 export function makeResolver(options: Options = {}): MetroResolver {

